src/sentry/workflow_engine/tasks.py

```diff
@@ -140,22 +182,121 @@ def process_workflows_event(
     )
     process_workflows(event_data)
 
     metrics.incr("workflow_engine.tasks.process_workflow_task_executed", sample_rate=1.0)
+
+
+def build_trigger_action_task_params(action, detector, event_data: WorkflowEventData):
+    """Build parameters for trigger_action.delay() call."""
+    event_id = None
+    activity_id = None
+    occurrence_id = None
+
+    if isinstance(event_data.event, GroupEvent):
+        event_id = event_data.event.event_id
+        occurrence_id = event_data.event.occurrence_id
+    elif isinstance(event_data.event, Activity):
+        activity_id = event_data.event.id
+
+    return {
+        "action_id": action.id,
+        "detector_id": detector.id,
+        "workflow_id": getattr(action, "workflow_id", None),
+        "event_id": event_id,
+        "activity_id": activity_id,
+        "group_id": event_data.event.group_id,
+        "occurrence_id": occurrence_id,
+        "group_state": event_data.group_state,
+        "has_reappeared": event_data.has_reappeared,
+        "has_escalated": event_data.has_escalated,
+        "workflow_env_id": event_data.workflow_env.id if event_data.workflow_env else None,
+    }
+
+
+@instrumented_task(
+    name="sentry.workflow_engine.tasks.trigger_action",
+    queue="workflow_engine.trigger_action",
+    acks_late=True,
+    default_retry_delay=5,
+    max_retries=3,
+    soft_time_limit=25,
+    time_limit=30,
+    silo_mode=SiloMode.REGION,
+    taskworker_config=config.TaskworkerConfig(
+        namespace=namespaces.workflow_engine_tasks,
+        processing_deadline_duration=30,
+        retry=Retry(
+            times=3,
+            delay=5,
+        ),
+    ),
+)
+def trigger_action(
+    action_id: int,
+    detector_id: int,
+    workflow_id: int,
+    event_id: str | None,
+    activity_id: int | None,
+    group_id: int,
+    occurrence_id: str | None,
+    group_state: GroupState,
+    has_reappeared: bool,
+    has_escalated: bool,
+    workflow_env_id: int | None,
+) -> None:
+
+    # XOR check to ensure exactly one of event_id or activity_id is provided
+    assert (event_id is not None) != (
+        activity_id is not None
+    ), "Exactly one of event_id or activity_id must be provided"
+
+    # Fetch the action and detector
+    action = Action.objects.annotate(workflow_id=Value(workflow_id)).get(id=action_id)
+    detector = Detector.objects.get(id=detector_id)
+
+    project_id = detector.project_id
+
+    if event_id is not None:
+        event_data = build_workflow_event_data(
+            project_id=project_id,
+            event_id=event_id,
+            group_id=group_id,
+            occurrence_id=occurrence_id,
+            group_state=group_state,
+            has_reappeared=has_reappeared,
+            has_escalated=has_escalated,
+            workflow_env_id=workflow_env_id,
+        )
+    else:
+        # Here, we probably build the event data from the activity
+        raise NotImplementedError("Activity ID is not supported yet")
```

Comment:
fyi: https://github.com/getsentry/sentry/blob/master/src/sentry/notifications/notification_action/utils.py#L27-L32 -- what will be interesting is to figure out how to map all the data in an activity to what is needed for a notification for metric issues since there are custom handlers setup for those.

we'll want to fall through to this case in the near-ish future, so we can disable workflow notifications (notifications from the issue platform) and allow any notification to filter through this system